internal/resolution/manifest/maven.go

```diff
@@ -837,7 +844,7 @@ func writeString(enc *xml.Encoder, raw, tag string, values map[string]string) er
 		if err != nil {
 			return err
 		}
-		if tt, ok := token.(xml.StartElement); ok && tt.Name.Local != tag {
+		if tt, ok := token.(xml.StartElement); ok && noInner(tt.Name.Local) {
```

Comment:
All the uses of `noInner` above seems to be checking a string against a blocklist. Can we simplify `noInner` to just be a slice instead? Then inside `writeString`, we just do `slices.Contains`.

Reply:
Alternatively, since this seems a bit brittle (we need to know all the cases where we don't want to trim out, is this something we can more reliably trim out by tracking nesting levels? 

e.g.. only replace values 1 level down.

Reply:
considering Maven POM models is quite well defined so we have confidence in knowing all the tag names and their levels except for property names (they can be any text) so I think it should be fine.

Reply:
now I rethink this - maybe this should be an improvement here (property names may be texts that we do not allow for trimming), though I need to investigate and understand more XML...